app/services/edition_withdrawer.rb

```diff
@@ -16,6 +33,25 @@ def past_participle
 
 private
 
+  def populate_with_previous_withdrawal
+    @edition.unpublishing.explanation = previous_withdrawal.explanation
+    @edition.unpublishing.unpublished_at = previous_withdrawal.unpublished_at
+  end
+
+  def creating_new_record?
+    @options[:unpublishing].present?
+  end
+
+  def use_previous_withdrawal?
+    @use_previous_withdrawal.present? && @use_previous_withdrawal != "new"
```

Comment:
Is this `use_previous_withdrawal != "new"` part of user input? if so I wonder if we want to deal with that at the controller level and just have a boolean in. I had assumed thus far that we were considering @use_previous_withdrawal as a boolean rather than a string or an id. Might be best to separate them into just a previous_withdrawal_id that is an integer or nil, rather than having user input go deeper into the classes.